examples/tensorflow-unit-testing/example_test.py

```diff
@@ -0,0 +1,115 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest.mock import patch
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from tensorflow.python.keras import testing_utils
+from tensorflow.python.keras import initializers
+
+import example
+
+
+class LinearBlockFullTest(tf.test.TestCase):
+    def test_basic(self):
+        testing_utils.layer_test(example.LinearBlockFull, input_shape=(4, 32))
+
+    def test_output(self):
+        batch_size, dim, output_dim = (3, 4, 2)
+        testing_utils.layer_test(
+            example.LinearBlockFull,
+            kwargs={'units': output_dim},
+            input_data=np.ones((batch_size, dim)),
+            expected_output_dtype='float32'
+        )
+
+
+class LinearBlockTest(tf.test.TestCase):
+    def test_shape_default(self):
+        x = np.ones((4, 32))
+        layer = example.LinearBlock()
+        output = layer(x)
+        self.assertAllEqual(output.shape, (4, 32))
+
+    def test_shape(self):
+        batch_size, input_dim, output_dim = (4, 15, 15)
+        x = np.ones((batch_size, input_dim))
+        layer = example.LinearBlock(output_dim)
+        output = layer(x)
+        self.assertAllEqual(output.shape, (batch_size, output_dim))
+
+    def test_shape_multidim(self):
+        batch_size, input_dim, output_dim = (4, 15, 15)
+        x = np.ones((batch_size, 7, input_dim))
+        layer = example.LinearBlock(output_dim)
+        output = layer(x)
+        self.assertAllEqual(output.shape, (batch_size, 7, output_dim))
+
+    def test_output(self):
+        batch_size, input_dim, output_dim = (3, 4, 4)
+        x = np.ones((batch_size, input_dim))
+        layer = example.LinearBlock(output_dim)
+        output = layer(x)
+        expected_output = np.array([[0.1250, -0.0576, 0.0513, -0.0305]]*batch_size)
+        self.assertAllClose(output, expected_output, atol=1e-4)
+
+    @patch.object(initializers, 'get', lambda _: tf.compat.v1.keras.initializers.Ones)
+    def test_output_ones(self):
+        batch_size, input_dim, output_dim = (3, 4, 4)
+        x = np.ones((batch_size, batch_size, input_dim, output_dim))
+        layer = example.LinearBlock(output_dim)
+        output = layer(x)
+        expected_output = np.ones((batch_size, output_dim)) * (batch_size, input_dim, output_dim + 1)
+        self.assertAllClose(output, expected_output, atol=1e-4)
+
+
+class ExampleModelTest(tf.test.TestCase):
+    def _get_data(self):
+        dataset_path = tf.keras.utils.get_file(
+            'auto-mpg.data',
+            'http://archive.ics.uci.edu/ml/machine-learning-databases/auto-mpg/auto-mpg.data')
+        column_names = ['MPG','Cylinders','Displacement','Horsepower','Weight',
```

Comment:
nit: Commas should be followed by a space. Surprised your linter didn't catch this.
Some other things my linter has complained about:
- Some lines exceed 80 chars
- Would be nice to have more docstrings.

Reply:
fixed